frontend/src/metabase/visualizations/components/settings/ChartSettingInput.tsx

```diff
@@ -20,11 +20,13 @@ interface ChartSettingInputProps {
   value: string;
   onChange: (value: string) => void;
   id?: string;
+  onChangeSettings: () => void;
 }
 
 const ChartSettingInput = ({
   value,
   onChange,
+  onChangeSettings,
```

Comment:
Adding onChgangeSettings so that spreading props no longer causes console errors around onChangeSettings being applied to an `<input>` element

Reply:
This is a good change, though the future reader would not know this specific reason and could have removed the prop thinking it was not used.

What about explicitly [omitting](https://underscorejs.org/#omit) it from the props instead? like
```ts
_.omit(props, "onChangeSettings")
```

or even name the prop itself
```ts
_.omit(props, ...nonInputProps)

const nonInputProps = ["onChangeSettings"]
```

p.s. I'm not certain about the name, but you get the idea.

Reply:
Yeah, that's fair. It felt kinda silly doing it the way I did, though I wish we could do stuff like omit *without* underscore.

Reply:
> I wish we could do stuff like omit without underscore.

Introducing [ClojureScript](https://clojurescript.org/)!!! Then we could use the [native `dissoc`](https://cljs.github.io/api/cljs.core/dissoc).